src/sync.rs

```diff
@@ -1,15 +1,38 @@
+#[cfg(all(not(feature = "std"), loom))]
+compile_error!("loom requires the standard library");
+
+//loom support
+
 #[cfg(loom)]
 pub(crate) mod atomic {
-    pub(crate) use loom::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use loom::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
```

Comment:
Loom's `Ordering` is [just a re-export](https://docs.rs/loom/0.5.2/src/loom/sync/atomic/mod.rs.html#16) of `core`'s `Ordering`, so we don't need to include that in the mods here. It can just always be consumed from `core`.